base/ryu/shortest.jl

```diff
@@ -0,0 +1,362 @@
+@inline function writeshortest(buf::Vector{UInt8}, pos, x::T,
```

Comment:
as the only function which uses `unsafe` operations, should we add some bounds-checking for sanity?